code/modules/projectiles/guns/energy/mounted.dm

```diff
@@ -1,26 +1,78 @@
-/obj/item/gun/energy/e_gun/advtaser/mounted
-	name = "mounted taser"
-	desc = "An arm mounted dual-mode weapon that fires electrodes and disabler shots."
+
+/obj/item/gun/energy/mounted
+	name = "mounted e-gun"
+	desc = "The parent of all mounted energy guns"
 	icon = 'icons/obj/items_cyborg.dmi'
-	icon_state = "taser"
-	inhand_icon_state = "armcannonstun4"
+	inhand_icon_state = "armcannonlase"
+	ammo_x_offset = 1
+	shaded_charge = 1
 	force = 5
 	selfcharge = 1
+	trigger_guard = TRIGGER_GUARD_ALLOW_ALL
 	can_flashlight = FALSE
-	trigger_guard = TRIGGER_GUARD_ALLOW_ALL // Has no trigger at all, uses neural signals instead
 
-/obj/item/gun/energy/e_gun/advtaser/mounted/dropped()//if somebody manages to drop this somehow...
+/obj/item/gun/energy/mounted/dropped()
 	..()
 
-/obj/item/gun/energy/laser/mounted
+/obj/item/gun/energy/mounted/laser
 	name = "mounted laser"
 	desc = "An arm mounted cannon that fires lethal lasers."
-	icon = 'icons/obj/items_cyborg.dmi'
-	icon_state = "laser"
-	inhand_icon_state = "armcannonlase"
-	force = 5
-	selfcharge = 1
-	trigger_guard = TRIGGER_GUARD_ALLOW_ALL
+	icon_state = "laser_cyborg"
+	ammo_type = list(/obj/item/ammo_casing/energy/lasergun)
 
-/obj/item/gun/energy/laser/mounted/dropped()
-	..()
+/obj/item/gun/energy/mounted/taser
+	name = "mounted taser"
+	desc = "An arm mounted dual-mode weapon that fires electrodes and disabler shots."
+	icon_state = "taser"
+	inhand_icon_state = "armcannonstun4"
+	ammo_type = list(/obj/item/ammo_casing/energy/electrode, /obj/item/ammo_casing/energy/disabler)
+	ammo_x_offset = 2
+
+/obj/item/gun/energy/mounted/leachgun
+	name = "mounted leach-rifle"
+	desc = "Exchange your own vitality for awesome power!\n Use in-hand to begin the recharge proccess."
+	icon_state = "leach_gun"
+	inhand_icon_state = "armcannonleach"
+	ammo_type = list(/obj/item/ammo_casing/energy/laser/leach)
+	selfcharge = 0
+	var/cooldown = 0
+
+/obj/item/gun/energy/mounted/leachgun/attack_self(mob/living/user)
+	if(cooldown > world.time)
+		to_chat(user, "<span class='userwarning'>The [src] is still getting ready to recharge again.</span>")
+		return
+	if(src.cell.percent() == 100)
+		to_chat(user, "<span class='userwarning'>The [src] is already fully charged.</span>")
+		return
+	cooldown = (world.time + 20 SECONDS)
```

Comment:
does this really need such a long cooldown? what if someone bumps you during the do_after?

Reply:
if the gun is fully spent it will take 19 seconds to recharge. 3 seconds per 30% missing charge and then 7 seconds once the process is over. this is to stop multiple charges to be queued up

Reply:
ill clean this up and remove the cooldown